### PR TITLE
fix: VitaminTextField rightView animation

### PR DESCRIPTION
**Pull request by bcleton**
## Changes description
We need to set the rightView frame when updating the icon to prevent animation from happening.

## Checklist

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch**. Please, don't request directly from your main!
- [x] Check commits & PR names matches our requested structure. It must follow the https://www.conventionalcommits.org pattern.
- [x] Check your code additions will fail neither code linting checks.
- [x] I have reviewed the submitted code.
- [x] I have tested on an iPhone device/simulator.
- [ ] I have tested on an iPad device/simulator.

## Does this introduce a breaking change?
- No

#### iPhone

animation

### CODE REVIEW

**Comment by florentlotthepro** [comment]
Thanks @bcleton for your contribution to Vitamin iOS!
Could you give us more information about the issue, please?
Your gif is the issue that you encounter?
Can we replicate it in the showcase app? To approve your fix.

**Comment by bcleton** [comment]
Yes the gif is the issue i encounter. You can see the animation of the icon (left to right) when i focus the second VitaminTextField.
To replicate the issue you need to have this structure inside a storyboard (i am adding the VitaminTextFIeld directly inside the Storyboard). All the TextFIelds are inside the StackView "FIelds".
Capture d’écran 2022-11-09 à 10 30 33
You then need to set `vitaminTextField.activeOnEditing = true`. This bug is quite hard to reproduce and doesn't happen every time you focus on a VitaminTextField. You have to do severals tries.

I found this fix on this stackoverflow:
https://stackoverflow.com/questions/18853972/how-to-stop-the-animation-of-uitextfield-rightview

**Comment by florentlotthepro** [comment]
@bcleton Thanks, for the details.

File: Sources/VitaminUIKit/Components/TextField/VitaminTextField.swift [hunk 1]
```diff
@@ -648,6 +648,7 @@ public class DesignableUITextField: UITextField {
                 imageView.isUserInteractionEnabled = true
             }
             rightView = imageView
+            rightView?.frame = self.rightViewRect(forBounds: self.bounds)
```
Comment (florentlotthepro) [inline Sources/VitaminUIKit/Components/TextField/VitaminTextField.swift:651]: Could you just add a little comment above your line to explain why we do this?
Something like `Due to a keyboard animation issue, we need to fix the frame of the right icon`.

**Review by florentlotthepro — APPROVED** [review]

File: Sources/VitaminUIKit/Components/TextField/VitaminTextField.swift (hunk 1 above)
Comment (bcleton) [reply Sources/VitaminUIKit/Components/TextField/VitaminTextField.swift]: Comment added 👍

Comment (florentlotthepro) [reply Sources/VitaminUIKit/Components/TextField/VitaminTextField.swift]: Thanks

**Comment by florentlotthepro** [comment]
I let someone else check the pull request and I let you sign our Contributor License Agreement (see CLAassistant message) to merge this PR.

**Review by florentlotthepro — APPROVED** [review]

**Review by daniel-dumortier — APPROVED** [review]
Ok for me, thanks for contribution !

**Pull request merged**